crates/uv-client/src/registry_client.rs

```diff
@@ -120,7 +120,7 @@ impl RegistryClientBuilder {
         // Initialize the base client.
         let client = self.client.unwrap_or_else(|| {
             // Load the TLS configuration.
-            let tls = tls::load(if self.native_tls {
+            let tls = tls::load(if self.native_tls || env::var("SSL_CERT_FILE").is_ok() {
```

Comment:
Should we instead change `tls::load` to add a certificate based on `SSL_CERT_FILE`? I think we would just call `root_cert_store.add` with the result of this: https://github.com/astral-sh/uv/pull/2351/files#diff-dcc1f69d132524500a0dcf217aa6db521517536daaf9364f3085961626fe722cR120

Reply:
I was thinking more towards delegating the parsing of `SSL_CERT_FILE` to [rustls-native-certs](https://github.com/rustls/rustls-native-certs/blob/main/src/lib.rs#L73) as we already do when we combine (`--native-tls` with `SSL_CERT_FILE`).

The `SSL_CERT_FILE` is actually a CA bundle, meaning it can contain more than one cert. I believe we'd have to load each entry, get a vector of Certificates, and them add them to the builder one by one and handle potential incompatible cert errors along the way (we get this for free with rustls-native-certs).
I did see an API reqwest to [load a pem bundle](https://docs.rs/reqwest/latest/reqwest/tls/struct.Certificate.html#method.from_pem_bundle), but didn't see one to add them all at once, so we'd need to call `add_root_certificate` for each one in a loop as we build client.

Sidenote, I've just tested this current PR with creating my own self-signed CA's and set `SSL_CERT_FILE` to this and can confirm this solution would effectively delegate the parsing of `SSL_CERT_FILE` to `rustls-native-certs` properly (as we already do with the combination of `--native-tls` with `SSL_CERT_FILE`)

Reply:
If we're concerned with a user providing an invalid `SSL_CERT_FILE` path, how about a nice warning?

```rust
            let ssl_cert_file_exists = env::var_os("SSL_CERT_FILE")
                .map_or(false, |path| {
                    let path_exists = Path::new(&path).exists();
                    if !path_exists {
                        warn_user_once!("Ignoring invalid value from environment for SSL_CERT_FILE. File path {} does not exists.", path.to_string_lossy());
                    }
                    path_exists
                });

            let tls = tls::load(if self.native_tls || ssl_cert_file_exists {
                Roots::Native
            } else {
                Roots::Webpki
            })
```

Reply:
I think that seems reasonable.

Reply:
Added in 85d92e8